### PR TITLE
bt-662: Connect badges

### LINKED ISSUE #662: Replace mocked badges with real data
**Acceptance criteria:**

We should connect the client to API

- Talent should be able to select badges “Communication score”, “Working with the team score”. “Project score”, “Lecture score”, “Level of English” should be displayed by default.
- All badges should display the score.
- When talent switches between steps, chosen badges should be saved.
- Chosen badges should be displayed on the Talent card/profile

### LINKED ISSUE #678: [BUG] "Your average project score" checkbox is not displayed as selected by default
**Description:**
"Your average project score" checkbox is not displayed as selected by default

**Preconditions:**
The BSA talent website is open

**Environment:**
OS Windows 11
Google Chrome 116.0

**Steps to reproduce:**
1. Sign up as a talent
2. Go to step 02
3. Pay attention to the Your average project score" checkbox

**Actual Result:**
"Your average project score" checkbox is not displayed as selected by default

**Expected Result:**
"Your average project score" checkbox must be displayed as selected by default

**Visual Proof (screenshots, videos, text):**
Screenshot_8

**Severity:**
Minor (S4)

### LINKED ISSUE #806: [BUG] Error message is displayed after going to step 03
**Description:**
Error message is displayed after going to step 03

**Actual Result:**
When filling in step 2 
click on the next button 
Error message is displayed after going to step 03

**Expected Result:**
Error message is not displayed after going to step 03

**Visual Proof (screenshots, videos, text):**
Screenshot_29

**Severity:**
Major (S3)

### PR DESCRIPTION
**Pull request by lavandosya**
After the merge, please, do next things:
1. run the migration
2. run seeds

### CODE REVIEW

File: shared/src/helpers/get-badge-icon/get-badge-icon.ts [hunk 1]
```diff
@@ -3,9 +3,12 @@ import { BsaBadgesTypeEnum } from '../../index.js';
...
     const values = {
-        [BsaBadgesTypeEnum.BSA_SCORE]: 'headphones',
-        [BsaBadgesTypeEnum.PERSONALITY]: 'account-multiple-outline',
-        [BsaBadgesTypeEnum.SOFT_SKILLS]: 'account-star-outline',
+        [BsaBadgesTypeEnum.ENGLISH_LEVEL]: 'headphones',
+        [BsaBadgesTypeEnum.PROJECT_SCORE]: 'headphones',
+        [BsaBadgesTypeEnum.LECTURE_SCORE]: 'headphones',
+        [BsaBadgesTypeEnum.BEST_LECTURE_SCORE]: 'headphones',
+        [BsaBadgesTypeEnum.TEAM_SCORE]: 'headphones',
+        [BsaBadgesTypeEnum.COMMUNICATION_SCORE]: 'headphones',
```
Comment (MaksymGavatuk) [inline shared/src/helpers/get-badge-icon/get-badge-icon.ts:11]: Why we have the same icons for badges ? The same as design ?

File: backend/src/bundles/lms-data/helpers/map-talent-badges.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+};
+
+const badgeTypeToLMSProperty = {
+    [BsaBadgesTypeEnum.ENGLISH_LEVEL]: 'english',
+    [BsaBadgesTypeEnum.LECTURE_SCORE]: 'averageLectureScore',
+    [BsaBadgesTypeEnum.BEST_LECTURE_SCORE]: 'grade',
+    [BsaBadgesTypeEnum.PROJECT_SCORE]: 'averageProjectScore',
+    [BsaBadgesTypeEnum.COMMUNICATION_SCORE]: 'communication_result',
+    [BsaBadgesTypeEnum.TEAM_SCORE]: 'team_interaction',
+};
```
Comment (MaksymGavatuk) [inline backend/src/bundles/lms-data/helpers/map-talent-badges.ts]: Move to mappers

File: backend/src/bundles/lms-data/helpers/map-talent-badges.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+const defaultShownBadges = {
+    [BsaBadgesTypeEnum.ENGLISH_LEVEL]: true,
+    [BsaBadgesTypeEnum.LECTURE_SCORE]: true,
+    [BsaBadgesTypeEnum.BEST_LECTURE_SCORE]: false,
+    [BsaBadgesTypeEnum.PROJECT_SCORE]: true,
+    [BsaBadgesTypeEnum.COMMUNICATION_SCORE]: false,
+    [BsaBadgesTypeEnum.TEAM_SCORE]: false,
+};
```
Comment (MaksymGavatuk) [inline backend/src/bundles/lms-data/helpers/map-talent-badges.ts]: constants

File: backend/src/bundles/lms-data/helpers/map-talent-badges.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+const DECIMAL_PLACES = 2;
+
+const getRandomScore = (): number => {
+    const randomValue = MOCK_SCORE + Math.random() * (MAX_SCORE - MOCK_SCORE);
+    return Number(randomValue.toFixed(DECIMAL_PLACES));
+};
+
+const isNumber = (value: unknown): value is number => {
+    return typeof value === 'number' && !Number.isNaN(value);
+};
```
Comment (MaksymGavatuk) [inline backend/src/bundles/lms-data/helpers/map-talent-badges.ts]: helpers

File: backend/src/bundles/talent-badges/talent-badge.entity.ts [hunk 5]
```diff
@@ -17,6 +19,8 @@ class TalentBadgeEntity implements Entity {
 
     private 'userDetailsId': string;
 
+    private 'badge': BSABadgeEntity | null;
```
Comment (MaksymGavatuk) [inline backend/src/bundles/talent-badges/talent-badge.entity.ts]: Why we need it in entity just create type for it and map to type

**Review by MaksymGavatuk — CHANGES_REQUESTED** [review]

File: shared/src/helpers/get-badge-icon/get-badge-icon.ts (hunk 1 above)
Comment (lavandosya) [reply shared/src/helpers/get-badge-icon/get-badge-icon.ts:11]: Yes, they are the same in the figma

**Comment by MaksymGavatuk** [comment]
Leave steps which you described in channel like desc of this pr

**Review by nikita-remeslov — APPROVED** [review]

**Comment by nika-sirak** [comment]
I got an error

Uploading Screen Recording 2023-09-29 at 18.20.18.mov…

**Review by nika-sirak — APPROVED** [review]

**Pull request merged**